### PR TITLE
Add new E2E error for expected UTD

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Add a new E2E error type for [`Expected UTD` ](https://github.com/element-hq/element-meta/issues/2313)
Also changed the json schema from `enum` to `oneOf` in order to put some comments

### CODE REVIEW

File: types/kotlin/Error.kt [hunk 1]
```diff
@@ -39,16 +39,3 @@ enum class Domain {
 enum class EventName {
     Error
 }
-
-enum class Name {
```
Comment (richvdh) [inline types/kotlin/Error.kt:43]: is this going to break any existing code?

File: schemas/Error.json [hunk 2]
```diff
@@ -15,17 +15,18 @@
...
-        "VoipInviteTimeout",
-        "VoipIceTimeout",
-        "VoipUserMediaFailed",
-        "ToDeviceFailedToDecrypt"
+      "oneOf": [
+        {"const": "OlmIndexError", "description": "E2E domain error. The room key is known but is ratcheted (index > 0)."},
+        {"const": "OlmKeysNotSentError", "description": "E2E domain error. Generic unknown inbound group session error."},
+        {"const": "OlmUnspecifiedError", "description": "E2E domain error. Any other decryption error (missing field, format errors...)."},
+        {"const": "UnknownError", "description": "E2E domain error. Decryption fails due to unknown error."},
+        {"const": "HistoricalMessage", "description": "E2E domain error. Decryption fails for historical message and there is no backup."},
```
Comment (richvdh) [inline schemas/Error.json:23]: Can we define what "historical message" means a bit better?

```suggestion
        {"const": "HistoricalMessage", "description": "E2E domain error. Decryption failed for a message sent before the device logged in, and key backup is not enabled."},
```

Comment (richvdh) [inline schemas/Error.json:23]: These should be `E2EE domain` for consistency with the `domain` enum

File: schemas/Error.json [hunk 3]
```diff
@@ -15,17 +15,18 @@
...
-        "VoipIceFailed",
-        "VoipInviteTimeout",
-        "VoipIceTimeout",
-        "VoipUserMediaFailed",
-        "ToDeviceFailedToDecrypt"
+      "oneOf": [
+        {"const": "OlmIndexError", "description": "E2E domain error. The room key is known but is ratcheted (index > 0)."},
+        {"const": "OlmKeysNotSentError", "description": "E2E domain error. Generic unknown inbound group session error."},
+        {"const": "OlmUnspecifiedError", "description": "E2E domain error. Any other decryption error (missing field, format errors...)."},
+        {"const": "UnknownError", "description": "E2E domain error. Decryption fails due to unknown error."},
```
Comment (richvdh) [inline schemas/Error.json:22]:
```suggestion
        {"const": "UnknownError", "description": "E2E domain error. Decryption failed due to unknown error."},
```

**Review by richvdh — CHANGES_REQUESTED** [review]
I think there should be typescript updates too?

File: types/kotlin/Error.kt (hunk 1 above)
Comment (BillCarsonFr) [reply types/kotlin/Error.kt:43]: kotlin is not used, the apps are using kotlin2

Comment (richvdh) [reply types/kotlin/Error.kt:43]: so can we get rid of `kotlin` altogether? Maybe a followup task.

**Review by richvdh — APPROVED** [review]
lgtm

File: types/kotlin/Error.kt (hunk 1 above)
Comment (BillCarsonFr) [reply types/kotlin/Error.kt:43]: https://github.com/matrix-org/matrix-analytics-events/issues/94

**Review by bmarty — APPROVED** [review]
OK, thanks!

**Pull request merged**